src/auth/AuthApiClient.ts

```diff
@@ -71,22 +78,38 @@ export class AuthApiClient {
     });
   }
 
-  async uploadUserProfileImage(image: any, token: string) {
+  async uploadUserProfileImage(image: UploadableFile, token: string) {
     const response = await fetch(`${this.endpoint}/auth/profile/image`, {
       method: "POST",
       headers: {
         Authorization: token,
+        "Content-Type": image.type,
+        "Content-Length": image.size.toString(),
       },
+      body: image.buffer.buffer,
     });
 
-    const data = await response.json();
-
-    await fetch(data.url, {
-      method: "PUT",
-      body: image,
-    });
-
-    return data;
+    if (!response.ok) {
+      if (response.status === 413) {
+        throw new FileTooLargeError();
+      }
+
+      if (response.status === 400) {
+        let data;
+        try {
+          data = await response.json<{ message?: string }>();
+        } catch {}
```

Comment:
I'm still not sold on this try/catch. If you look at possible exceptions for the `.json()` function, there's not really anything that could throw here. So in the rare case something fails here, javascript throws the error by itself and that should be just fine.